lib/vagrant-vbguest/installers/linux.rb

```diff
@@ -78,7 +78,7 @@ def running?(opts=nil, &block)
         opts = {
           :sudo => true
         }.merge(opts || {})
-        communicate.test('lsmod | grep vboxsf', opts, &block)
+        communicate.test('grep -qE "^vboxguest\s.+\s\([^\s]*O[^\s]*\)$" /proc/modules && test -e /lib/modules/`uname -r`/misc/vboxsf.ko', opts, &block)
```

Comment:
Have you had any chance to test this on older versions?

Reply:
I currently don't have an older version of virtualbox installed, if that's what you're asking. But (IIRC) the guest additions have placed the built modules in the `misc` directory since forever.

I have also not tested on older vagrant versions (I have 2.2.4).

Reply:
This test doesn't work with CentOS 6.5 and VirtualBox 6.0.10 as the line in the /proc/modules is the following which isn't matched by this regex:

`vboxguest 337227 2 vboxsf, Live 0x0000000000000000`